java/core/src/main/scala/io/pdal/PackedPoints.scala

```diff
@@ -0,0 +1,66 @@
+package io.pdal
+
+import java.nio.ByteBuffer
+import java.util
+
+import scala.collection.JavaConversions._
+
+/**
+  * PackedPoints abstraction to work with packed point(s) in JVM memory.
+  * SizedDimType contains size and offset for a particular packed point with the current set of dims.
+  **/
+case class PackedPoints(bytes: Array[Byte], dimTypes: util.Map[String, SizedDimType], metadata: String = "", schema: String = "") {
+  def pointSize: Long = dimTypes.values.map(_.size).sum
+  def length: Int = (bytes.length / pointSize).toInt
+  def isPoint: Boolean = length == pointSize
+  def dimSize(dim: SizedDimType) = dimTypes(dim.dimType.id).size
+  def dimSize(dim: DimType) = dimTypes(dim.id).size
+  def dimSize(dim: String) = dimTypes(dim).size
+  def findDimType(dim: String) = dimTypes(dim).dimType
+  def findSizedDimType(dim: String) = dimTypes(dim)
+
+  /**
+    * Reads a packed point by point id from a set of packed points.
+    */
+  def get(i: Int): Array[Byte] = {
+    val pointSize = this.pointSize
+    if (isPoint) bytes
+    else {
+      val from = (i * pointSize).toInt
+      val to = {
+        val t = (from + pointSize).toInt
+        if (t > bytes.length) bytes.length else t
+      }
+
+      bytes.slice(from, to)
+    }
+  }
+
+  def get(idx: Int, dim: SizedDimType): ByteBuffer = get(idx, dim.dimType.id)
+  def get(idx: Int, dim: DimType): ByteBuffer = get(idx, dim.id)
+  def get(idx: Int, dim: String): ByteBuffer = ByteBuffer.wrap(get(get(idx), dim))
```

Comment:
We should have versions like `getInt`, `getByte` etc to do the ByteBuffer => Value conversions for the user.